packages/repository/src/repositories/relation.repository.ts

```diff
@@ -53,10 +97,56 @@ export class DefaultHasManyEntityCrudRepository<
     );
   }
 
-  async find(filter?: Filter | undefined, options?: Options): Promise<T[]> {
+  async find(filter?: Filter, options?: Options): Promise<T[]> {
     return await this.targetRepository.find(
       constrainFilter(filter, this.constraint),
       options,
     );
   }
+
+  async patch(entity: DataObject<T>, options?: Options): Promise<boolean> {
+    return await this.targetRepository.update(
+      constrainDataObject(entity, this.constraint),
+      options,
+    );
+  }
+
+  async delete(entity: DataObject<T>, options?: Options): Promise<boolean> {
+    return await this.targetRepository.delete(
+      constrainDataObject(entity, this.constraint),
+      options,
+    );
+  }
+
+  async replace(entity: DataObject<T>, options?: Options): Promise<boolean> {
+    return await this.targetRepository.replaceById(
+      entity.getId(),
+      constrainDataObject(entity, this.constraint),
+      options,
+    );
+  }
+
+  async createAll(entities: DataObject<T>[]): Promise<T[]> {
+    entities = entities.map(e => constrainDataObject(e, this.constraint));
+    return await this.targetRepository.createAll(entities);
+  }
+
+  async deleteAll(where?: Where, options?: Options): Promise<number> {
+    return await this.targetRepository.deleteAll(
+      constrainWhere(where, this.constraint),
+      options,
+    );
+  }
+
+  async patchAll(
+    dataObject: DataObject<T>,
+    where?: Where,
+    options?: Options,
+  ): Promise<number> {
+    return await this.targetRepository.updateAll(
+      dataObject,
```

Comment:
I think you need to call `constrainDataObject`?

Test case:

- `Category hasMany products`, using `product.categoryId` as the FK.
- create releation repository to access products of a category with id `CATID`
- call `patchAll({categoryId: 123})` - this moves the products from CATID to category 123, I think it's not allowed.

Reply:
@bajtos Good catch! I added the test case in unit test.